test/testsuite.h

```diff
@@ -42,6 +42,8 @@ class TestFixture : public ErrorLogger {
     std::string mTemplateLocation;
     std::string mTestname;
 
+    static std::string writestr(const std::string& str, bool gccStyle = false);
```

Comment:
The changes to this file come from #3056. The goal in that PR was to allow equality assertions for arbitrary types. This exposed the problem that we are sometimes asserting the equality of bools with ints (by implicitly converting to ints). Adding this overload causes those conversions to (correctly in my opinion) issue a warning.

Reply:
basically.. I prefer local functions in a cpp file. unless there are good reasons to declare them in the headers.. and right now I don't feel there are strong reasons to publish this writestr.

Reply:
writestr doesn't have to be exposed anymore in the header as far as I see.